pkg/integrations/v4/emitter/emitter.go

```diff
@@ -75,6 +75,23 @@ func (e *VersionAwareEmitter) Emit(definition integration.Definition, extraLabel
 		return err
 	}
 
+	// Agent creating the Host entity (and decorating it correctly in the backend) in secure forward with Custom Attributes: pkg/plugins/plugins_linux.go:46
+	// But in forward only there is no host entity and custom attributes are not being decorated.
+	// Here then we add CustomAttributes to extraLabels in case we are in that mode.
+	if e.aCtx.Config().IsForwardOnly {
+		extraLabelsCopy := make(map[string]string)
+		customAttributes := e.aCtx.Config().CustomAttributes.ToDataMap()
+
+		for k, v := range extraLabels {
+			extraLabelsCopy[k] = v
+		}
+		for k, v := range customAttributes {
+			extraLabelsCopy[k] = v
+		}
+
+		extraLabels = extraLabelsCopy
```

Comment:
I think that `extraLabelsCopy` is not necessary and that it can be simplified adding custom attributes to `extraLabels` as it is being overwritten in line 92
```go
	if e.aCtx.Config().IsForwardOnly {
		customAttributes := e.aCtx.Config().CustomAttributes.ToDataMap()

		for k, v := range customAttributes {
			extraLabels[k] = v
		}
	}
```

Reply:
That modifies the `extraLabels` variable from the caller of the `Emit` function and I wanted to leave the variable untouched.

In fact one of the tests gave a false positive if I remove the copy of the map because modified the expected while I did not even modified the test to compare with the custom attributes.

Reply:
but `extraLabels` is overwritten in line 92, or am I missing something? 🤔

Reply:
I understand know. I would have expected original `extraLabels` to have been changed in line 92 but it does not.